binding-macro/src/service.rs

```diff
@@ -87,81 +87,99 @@ pub fn gen_service_code(_: TokenStream, item: TokenStream) -> TokenStream {
-                        let payload: #list_read_payload = serde_json::from_str(ctx.get_payload())
-                                .map_err(|e| protocol::traits::BindingMacroError::JsonParse(e))?;
-                        let res = self.#list_read_ident(ctx, payload)?;
-                        let res_str = serde_json::to_string(&res).map_err(|e| protocol::traits::BindingMacroError::JsonParse(e))?;
-                        if res_str.as_str() == "null" {
-                            Ok("".to_owned())
+                        let payload_res: Result<#list_read_payload, _> = serde_json::from_str(ctx.get_payload());
+                        if payload_res.is_err() {
+                            return ServiceResponse::<String>::from_error(1, "service macro decode payload failed".to_owned());
+                        };
+                        let payload = payload_res.unwrap();
+                        let res = self.#list_read_ident(ctx, payload);
+                        if !res.is_error() {
+                            let mut data_json = serde_json::to_string(&res.succeed_data).unwrap_or_else(|e| panic!("service macro encode payload failed: {:?}", e));
+                            if data_json == "null" {
+                                data_json = "".to_owned();
+                            }
+                            ServiceResponse::<String>::from_succeed(data_json)
                         } else {
-                            Ok(res_str)
+                            ServiceResponse::<String>::from_error(res.code, res.error_message.clone())
                         }
                     },)*
                     #(#list_read_name_nonepayload => {
-                        let res = self.#list_read_ident_nonepayload(ctx)?;
-                        let res_str = serde_json::to_string(&res).map_err(|e| protocol::traits::BindingMacroError::JsonParse(e))?;
-                        if res_str.as_str() == "null" {
-                            Ok("".to_owned())
+                        let res = self.#list_read_ident_nonepayload(ctx);
+                        if !res.is_error() {
+                            let mut data_json = serde_json::to_string(&res.succeed_data).unwrap_or_else(|e| panic!("service macro encode payload failed: {:?}", e));
+                            if data_json == "null" {
+                                data_json = "".to_owned();
+                            }
+                            ServiceResponse::<String>::from_succeed(data_json)
                         } else {
-                            Ok(res_str)
+                            ServiceResponse::<String>::from_error(res.code, res.error_message.clone())
                         }
                     },)*
-                    _ => Err(protocol::traits::BindingMacroError::NotFoundMethod{ service: service.to_owned(), method: method.to_owned() }.into())
+                    _ => panic!("service macro not found method:{:?} of service:{:?}", method, service)
```

Comment:
Should not panic, method not found is normal.

Reply:
fixed:
```rust
_ => ServiceResponse::<String>::from_error(2, format!("not found method:{:?} of service:{:?}", method, service))
```